--- mk-108.1/next/eventc.h
+++ 07_kernel/v183.34/m68k/src/machdep/m68k/eventc.h
@@ -34,10 +34,10 @@
 #define _EVENTC_
 
 #if	KERNEL
-#import <next/cpu.h>
-#import <next/spl.h>
+#import <bsd/m68k/cpu.h>
+#import <bsd/m68k/spl.h>
 #import <kern/assert.h>
-#import <sys/time_stamp.h>
+#import <kern/time_stamp.h>
 
 extern volatile u_char		*eventc_latch, *eventc_h, *eventc_m, *eventc_l;
 extern volatile unsigned int	*event_high; /* high-order 32 bits of eventc */
