--- 07_kernel/src/mach/i386/vm_param.h	2026-10-01 12:33:15.721813795 +0900
+++ 08_build/runs/tools/s5p87-probe-stage-1/src/mach/i386/vm_param.h	2026-10-02 11:40:38.513826618 +0900
@@ -68,7 +68,7 @@
 #define VM_MIN_KERNEL_ADDRESS	((vm_offset_t) 0x00000000)
 #define VM_MAX_KERNEL_ADDRESS	((vm_offset_t) 0x40000000)
 
-#define KERNSTACK_SIZE		(2*I386_PGBYTES)
+#define KERNSTACK_SIZE		(I386_PGBYTES)
 #define INTSTACK_SIZE		(1*I386_PGBYTES)
 
 /*
