--- darwin01/kernel/machdep/i386/libc/memcpy.c	1999-03-17 01:55:35.000000000 +0900
+++ src/machdep/i386/libc/memcpy.c	2026-10-01 14:34:42.672892887 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /* Copyright (c) 1991 NeXT Computer, Inc.  All rights reserved.
  *
  *      File:   libc/i386/ansi/memcpy.c
@@ -98,6 +100,11 @@
 
 void *memcpy(void *, const void *, unsigned long);
 
+void bcopy(const void *src, void *dst, unsigned long ulen)
+{
+    (void) memcpy(dst, src, ulen);
+}
+
 void *memcpy(void *dst, const void *src, unsigned long ulen)
 {
     int len = ulen;
--- darwin01/kernel/machdep/i386/libc/memmove.c	1999-03-17 01:55:35.000000000 +0900
+++ src/machdep/i386/libc/memmove.c	2026-10-01 14:34:42.672892887 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /* Copyright (c) 1991 NeXT Computer, Inc.  All rights reserved.
  *
  *      File:   libc/i386/ansi/memmove.c
@@ -112,11 +114,6 @@
 
 void *memmove(void *, const void *, unsigned long);
 
-void bcopy(const void *src, void *dst, unsigned long ulen)
-{
-    (void) memmove(dst, src, ulen);
-}
-
 void ovbcopy(const void *src, void *dst, unsigned long ulen)
 {
     (void) memmove(dst, src, ulen);
