--- darwin01/kernel/ipc/ipc_entry.h	1999-03-17 01:54:00.000000000 +0900
+++ ipc/ipc_entry.h	2026-10-01 22:16:14.343172205 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -204,7 +206,6 @@
 
 /* Grow the table in a space */
 extern kern_return_t ipc_entry_grow_table(
-	ipc_space_t	space,
-	int		target_size);
+	ipc_space_t	space);
 
 #endif	/* _IPC_IPC_ENTRY_H_ */
