--- 01_resources/upstream/darwin01/kernel/mach/port.h	1999-03-17 01:55:06.000000000 +0900
+++ 07_kernel/src/mach/port.h	2026-10-01 23:02:26.583709453 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -179,6 +181,17 @@
 
 #define MACH_PORT_QLIMIT_DEFAULT	((mach_port_msgcount_t) 5)
 #define MACH_PORT_QLIMIT_MAX		((mach_port_msgcount_t) 16)
+typedef struct old_mach_port_status {
+	mach_port_t		mps_pset;	/* containing port set */
+/*mach_port_mscount_t*/natural_t mps_mscount;	/* make-send count */
+/*mach_port_msgcount_t*/natural_t mps_qlimit;	/* queue limit */
+/*mach_port_msgcount_t*/natural_t mps_msgcount;	/* number in the queue */
+/*mach_port_rights_t*/natural_t	mps_sorights;	/* how many send-once rights */
+/*boolean_t*/natural_t		mps_srights;	/* do send rights exist? */
+/*boolean_t*/natural_t		mps_pdrequest;	/* port-deleted requested? */
+/*boolean_t*/natural_t		mps_nsrequest;	/* no-senders requested? */
+} old_mach_port_status_t;
+
 
 typedef struct mach_port_limits {
 	mach_port_msgcount_t	mpl_qlimit;	/* number of msgs */
@@ -226,6 +239,6 @@
 #define PORT_NULL	((port_name_t) 0)	/* Used to denote no port; legal value */
 
 #define PORT_BACKLOG_DEFAULT	MACH_PORT_QLIMIT_DEFAULT
-#define PORT_BACKLOG_MAX	MACH_PORT_QLIMIT_MAX
+#define PORT_BACKLOG_MAX	16
 
 #endif	/* _MACH_PORT_H_ */
