--- 01_resources/upstream/darwin01/kernel/ipc/mach_msg.h	1999-03-17 01:54:08.000000000 +0900
+++ 07_kernel/src/ipc/mach_msg.h	2026-10-02 11:56:32.698503726 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-02: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -89,6 +91,9 @@
 extern void
 mach_msg_receive_continue();
 
+extern void
+mach_msg_continue();
+
 extern boolean_t
 mach_msg_interrupt(/* thread_t */);
 
