--- darwin01/kernel/ipc/ipc_kmsg.h	1999-03-17 01:54:02.000000000 +0900
+++ ipc/ipc_kmsg.h	2026-10-01 22:33:12.420825360 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -126,7 +128,6 @@
 	struct ipc_kmsg *ikm_next, *ikm_prev;
 	vm_size_t ikm_size;
 	ipc_marequest_t ikm_marequest;
-	security_id_t ikm_sender;
 	integer_t ikm_delta;
 	mach_msg_header_t ikm_header;
 } *ipc_kmsg_t;
@@ -171,7 +172,6 @@
 MACRO_BEGIN								\
 	(kmsg)->ikm_size = (size);					\
 	(kmsg)->ikm_marequest = IMAR_NULL;				\
-	(kmsg)->ikm_sender = ANONYMOUS_SECURITY_ID_VALUE;		\
 	(kmsg)->ikm_delta = 0;						\
 MACRO_END
 
@@ -179,7 +179,6 @@
 MACRO_BEGIN								\
 	assert((kmsg)->ikm_size == (size));				\
 	assert((kmsg)->ikm_marequest == IMAR_NULL);			\
-	(kmsg)->ikm_sender = ANONYMOUS_SECURITY_ID_VALUE;		\
 	(kmsg)->ikm_delta = 0;						\
 MACRO_END
 
