--- 01_resources/upstream/darwin01/kernel/kern/sched_prim.c	1999-03-17 01:54:45.000000000 +0900
+++ 08_build/runs/tools/s5p90-probe-stage-1/src/kern/sched_prim.c	2026-10-02 11:47:16.231108012 +0900
@@ -1189,11 +1189,6 @@
 
 	assert(ticks != 0);
 
-	if ((thread->state & TH_RUN) == 0)
-		thread->sleep_time += ticks;
-	else
-		thread->sleep_time = 0;
-
 	/*
 	 *	If asleep for more than 30 seconds forget all
 	 *	cpu_usage, else catch up on missed aging.
