--- 01_resources/upstream/nextmach/mk-108.1/rpc/pmap_kgetport.c	2026-09-11 10:02:49.614218873 +0900
+++ 07_kernel/src/bsd/rpc/pmap_kgetport.c	2026-10-03 11:52:33.390408588 +0900
@@ -36,6 +36,7 @@
 #import <sys/param.h>
 #import <sys/user.h>
 #import <sys/proc.h>
+#import <kern/thread.h>		/* plan 190: ISSIG uses current_thread()->_uthread (SDK sys/param.h) */
 static struct ucred cred;
 #define retries 4
 static struct timeval tottimeout = { 1, 0 };
