--- 08_build/runs/s5p343-mig3/out/audioReplyUser.c	2026-10-06 11:17:19.153821234 +0900
+++ 07_kernel/src/driverkit/libDriver/audioReplyUser.c	2026-10-06 11:17:36.764891288 +0900
@@ -198,7 +198,7 @@
 	InP->Head.msg_reply_port = PORT_NULL;
 	InP->Head.msg_id = 1700;
 
-	return msg_send(&InP->Head, MSG_OPTION_NONE, 0);
+	return msg_send(&InP->Head, SEND_TIMEOUT|SEND_SWITCH, 1000);
 }
 
 /* SimpleRoutine RecordedData */
@@ -370,5 +370,5 @@
 	InP->Head.msg_reply_port = PORT_NULL;
 	InP->Head.msg_id = 1701;
 
-	return msg_send(&InP->Head, MSG_OPTION_NONE, 0);
+	return msg_send(&InP->Head, SEND_TIMEOUT|SEND_SWITCH, 1000);
 }
