--- darwin01/kernel/ipc/ipc_marequest.c	1999-03-17 01:54:02.000000000 +0900
+++ ipc/ipc_marequest.c	2026-10-01 22:05:51.183396189 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /* 
  * Mach Operating System
  * Copyright (c) 1991,1990,1989 Carnegie Mellon University
@@ -456,7 +458,7 @@
 #endif	MACH_IPC_COMPAT
 
 	assert(soright != IP_NULL);
-	panic("ipc_marequest_destroy");
+	ipc_notify_msg_accepted(soright, name);
 }
 
 #if	MACH_IPC_DEBUG
