--- 01_resources/upstream/nextmach/mk-108.1/bsd/kern_xxx.c	2026-09-11 10:02:49.485217287 +0900
+++ 07_kernel/src/bsd/kern/kern_xxx.c	2026-10-03 12:35:03.681797094 +0900
@@ -171,6 +171,7 @@
 	};
 	char command[64];
 
+	command[0] = '\0';	/* plan 195 (authored; original 0x10b772) */
 	if (suser()) {
 		if (((struct a *)u.u_ap)->opt & RB_COMMAND)
 			u.u_error = copyinstr(((struct a *)u.u_ap)->command,
