--- 01_resources/upstream/darwin01/kernel/kern/ipc_tt.c	1999-03-17 01:54:33.000000000 +0900
+++ 07_kernel/src/kern/ipc_tt.c	2026-10-01 23:24:29.169804025 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /* 
  * Mach Operating System
  * Copyright (c) 1991,1990,1989,1988,1987 Carnegie Mellon University
@@ -416,60 +418,6 @@
 }
 
 /*
- *	Routine:	retrieve_task_self
- *	Purpose:
- *		Return a send right (possibly null/dead)
- *		for the task's user-visible self port.
- *	Conditions:
- *		Nothing locked.
- */
-
-ipc_port_t
-retrieve_task_self(task)
-	task_t task;
-{
-	ipc_port_t port;
-
-	assert(task != TASK_NULL);
-
-	itk_lock(task);
-	if (task->itk_self != IP_NULL)
-		port = ipc_port_copy_send(task->itk_sself);
-	else
-		port = IP_NULL;
-	itk_unlock(task);
-
-	return port;
-}
-
-/*
- *	Routine:	retrieve_thread_self
- *	Purpose:
- *		Return a send right (possibly null/dead)
- *		for the thread's user-visible self port.
- *	Conditions:
- *		Nothing locked.
- */
-
-ipc_port_t
-retrieve_thread_self(thread)
-	thread_t thread;
-{
-	ipc_port_t port;
-
-	assert(thread != ITH_NULL);
-
-	ith_lock(thread);
-	if (thread->ith_self != IP_NULL)
-		port = ipc_port_copy_send(thread->ith_sself);
-	else
-		port = IP_NULL;
-	ith_unlock(thread);
-
-	return port;
-}
-
-/*
  *	Routine:	retrieve_task_self_fast
  *	Purpose:
  *		Optimized version of retrieve_task_self,
