--- darwin01/kernel/mach/mach_param.h	1999-03-17 01:55:03.000000000 +0900
+++ mach/mach_param.h	2026-10-01 20:56:43.020231575 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -83,6 +85,6 @@
 
 /* Number of "registered" ports */
 
-#define TASK_PORT_REGISTER_MAX	3
+#define TASK_PORT_REGISTER_MAX	4
 
 #endif	/* _MACH_MACH_PARAM_H_ */
