--- darwin01/kernel/kern/thread_swap.c	1999-03-17 01:54:48.000000000 +0900
+++ kern/thread_swap.c	2026-10-01 17:53:10.120179254 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Mach Operating System
  * Copyright (c) 1991,1990,1989,1988,1987 Carnegie Mellon University
@@ -208,7 +210,6 @@
 void swapin_thread()
 {
 	stack_privilege(current_thread());
-	current_thread()->vm_privilege = TRUE;
 
 	swapin_thread_continue();
 	/*NOTREACHED*/
