--- darwin01/kernel/ipc/ipc_table.c	1999-03-17 01:54:07.000000000 +0900
+++ src/ipc/ipc_table.c	2026-10-01 16:20:33.748309760 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1995, 1994, 1993, 1992, 1991, 1990  
  * Open Software Foundation, Inc. 
@@ -103,7 +105,7 @@
 extern vm_map_t kalloc_map;
 
 ipc_table_size_t ipc_table_entries;
-unsigned int ipc_table_entries_size = 512;
+unsigned int ipc_table_entries_size = 128;
 
 ipc_table_size_t ipc_table_dnrequests;
 unsigned int ipc_table_dnrequests_size = 64;
@@ -144,8 +146,7 @@
 				index++;
 			}
 		}
-		if (incrsize < (PAGE_SIZE << 3))
-			incrsize <<= 1;
+		incrsize <<= 1;
 	}
 }
 
