--- darwin01/kernel/machdep/i386/idt.c	1999-03-17 01:55:25.000000000 +0900
+++ machdep/i386/idt.c	2026-10-01 17:31:37.677842340 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1992 NeXT Computer, Inc.
  *
@@ -434,18 +436,3 @@
 };
 
 idt_t *		idt = (idt_entry_t *)idt_pseudo;
-
-void
-idt_copy(
-    vm_offset_t		new_idt,
-    vm_size_t		length
-)
-{
-    if (length < (IDTSZ * sizeof (idt_entry_t)))
-	panic("idt_copy");
-
-    (void) memcpy((void *)new_idt, idt, IDTSZ * sizeof (idt_entry_t));
-    (vm_offset_t)idt = new_idt;
-
-    locate_idt((vm_offset_t) idt);
-}
