--- 01_resources/upstream/nextmach/mk-108.1/bsd/tty_conf.c	2026-09-11 10:02:49.491217361 +0900
+++ 07_kernel/src/bsd/kern/tty_conf.c	2026-10-08 14:51:59.626659040 +0900
@@ -58,20 +58,3 @@
 };
 
 int	nldisp = sizeof (linesw) / sizeof (linesw[0]);
-
-/*
- * Do nothing specific version of line
- * discipline specific ioctl command.
- */
-/*ARGSUSED*/
-nullioctl(tp, cmd, data, flags)
-	struct tty *tp;
-	char *data;
-	int flags;
-{
-
-#ifdef lint
-	tp = tp; data = data; flags = flags;
-#endif
-	return (-1);
-}
