--- darwin01/kernel/kern/task.h	1999-03-17 01:54:47.000000000 +0900
+++ kern/task.h	2026-10-01 20:56:43.007231436 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-01: see 07_kernel/MODIFICATIONS.md */
+
 /* 
  * Mach Operating System
  * Copyright (c) 1993-1988 Carnegie Mellon University
@@ -91,6 +93,7 @@
 	boolean_t	assign_active;	/* waiting for may_assign */
 
 	/* Garbage */
+	struct utask	*u_address;
 	struct proc	*proc;		/* corresponding process */
 
 	/* Task-wide thread context information */
