--- 01_resources/upstream/darwin01/kernel/mach/i386/vm_param.h	1999-03-17 01:55:01.000000000 +0900
+++ 07_kernel/src/mach/i386/vm_param.h	2026-10-02 11:42:12.321529156 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-02: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1992 NeXT Computer, Inc.
  *
@@ -68,7 +70,7 @@
 #define VM_MIN_KERNEL_ADDRESS	((vm_offset_t) 0x00000000)
 #define VM_MAX_KERNEL_ADDRESS	((vm_offset_t) 0x40000000)
 
-#define KERNSTACK_SIZE		(2*I386_PGBYTES)
+#define KERNSTACK_SIZE		(I386_PGBYTES)
 #define INTSTACK_SIZE		(1*I386_PGBYTES)
 
 /*
