--- 01_resources/upstream/darwin01/kernel/machdep/i386/dma_inline.h	1999-03-17 01:55:23.000000000 +0900
+++ 07_kernel/src/machdep/i386/dma_inline.h	2026-10-02 13:10:25.386557930 +0900
@@ -22,6 +22,8 @@
  * @APPLE_LICENSE_HEADER_END@
  */
 
+/* Modified 2026-10-02: see 07_kernel/MODIFICATIONS.md */
+
 /*
  * Copyright (c) 1992 NeXT Computer, Inc.
  *
@@ -42,7 +44,7 @@
 
 #import <machdep/i386/dma.h>
 #import <machdep/i386/dma_internal.h>
-#import <bsd/i386/param.h>
+#import <bsd/i386/machparam.h>
 
 #define DMA_DEBUG 0
 
