--- 01_resources/upstream/darwin01/kernel/kern/syscall_sw.c	1999-03-17 01:54:46.000000000 +0900
+++ 08_build/runs/tools/s5p77-probe-stage-1/src/kern/syscall_sw.c	2026-10-02 10:35:00.782708046 +0900
@@ -130,7 +130,7 @@
 	MACH_TRAP(kern_invalid, 0),		/* 19 */
 
 #if	MACH_IPC_COMPAT
-	MACH_TRAP_STACK(msg_send_trap, 4),	/* 20 */	/* obsolete */
+	MACH_TRAP(msg_send_trap, 4),	/* 20 */	/* obsolete */
 	MACH_TRAP_STACK(msg_receive_trap, 5),	/* 21 */	/* obsolete */
 	MACH_TRAP_STACK(msg_rpc_trap, 6),	/* 22 */	/* obsolete */
 #else	/* MACH_IPC_COMPAT */
@@ -139,8 +139,7 @@
 	MACH_TRAP(kern_invalid, 0),		/* 22 */
 #endif	/* MACH_IPC_COMPAT */
 	MACH_TRAP(kern_invalid, 0),		/* 23 */
-	MACH_TRAP_STACK(mach_msg_simple_trap, 5),
-						/* 24 */
+	MACH_TRAP(kern_invalid, 0),		/* 24 */
 	MACH_TRAP_STACK(mach_msg_trap, 7),	/* 25 */
 	MACH_TRAP(mach_reply_port, 0),		/* 26 */
 	MACH_TRAP(mach_thread_self, 0),		/* 27 */
@@ -149,11 +148,11 @@
 
 	MACH_TRAP(kern_invalid, 0),		/* 30 */
 	MACH_TRAP(kern_invalid, 0),		/* 31 */
-	MACH_TRAP_STACK(mach_msg_overwrite_trap, 9), /* 32 */
+	MACH_TRAP(kern_invalid, 0),		/* 32 */
 	MACH_TRAP(task_by_pid, 1),		/* 33 */
 	MACH_TRAP(kern_invalid, 0),		/* 34 */
 	MACH_TRAP(_lookupd_port, 1),		/* 35 */
-	MACH_TRAP(_lookupd_port1, 1),		/* 36 */
+	MACH_TRAP(kern_invalid, 0),		/* 36 */
 	MACH_TRAP(kern_invalid, 0),		/* 37 */
 	MACH_TRAP(kern_invalid, 0),		/* 38 */
 
